slimta/envelope.py

```diff
@@ -24,19 +24,32 @@
 
 """
 
-from __future__ import absolute_import
+from __future__ import absolute_import, unicode_literals
 
 import re
 import copy
-import cStringIO
 from email.message import Message
 from email.generator import Generator
-from email.parser import Parser, FeedParser
+from email.parser import Parser
+
+import six
+from six.moves import cStringIO
+from six import BytesIO
+
+if six.PY2:
+    from email.generator import Generator as BytesGenerator
+
+else:
+    from email.parser import BytesParser
+    from email.generator import BytesGenerator
```

Comment:
> For standard library differences, when a six convenience import is not available, I'd prefer to use try/excepts with the Py3 version first

Yes, IMHO it's less explicit (hint: you have to use a comment), and AttributeError/ImportError may be caused by something else.